syslog-ng-ctl/syslog-ng-ctl.c

```diff
@@ -408,9 +409,13 @@ slng_passwd(int argc, char *argv[], const gchar *mode, GOptionContext *ctx)
 {
   gulong buff_size = 255;
   gchar buff[buff_size+1];
+  gint remaining_unused_index = 0;
 
   if (g_strcmp0(mode,"add")==0)
     {
+      if (!credentials_key && credentials_remaining && credentials_remaining[remaining_unused_index])
```

Comment:
Factor out this to a function, if possible (try to name this expression, or, if it's not possible, then try to factor out a subexpression).

Reply:
Fixed